### ISSUE TITLE
Bug: `css/no-invalid-properties` throws TypeError with certain valid CSS

### ISSUE THREAD

**Issue by jgoz** [eslint/css#289]
### Environment

ESLint version: 9.37.0
@eslint/css version: 0.12.0
Node version: 22.18.0
npm version: N/A
Operating System: macOS

### Which language are you using?

stylesheet

### What did you do?

<details>
<summary>Configuration</summary>

```
// CSS rules
{
  files: ['**/*.css'],
  language: 'css/css',
  languageOptions: {
    tolerant: true,
  },
  plugins: { css },
  extends: [css.configs.recommended],
  rules: {
    'css/use-baseline': 'off',
    'css/no-invalid-properties': ['error', { allowUnknownVariables: true }],
  },
}
```
</details>

```css
@supports (color: color(display-p3 1 1 1)) {
  @media (color-gamut: p3) {
    & .my-class {
      --my-color: oklch(73.5% 0.1192 254.8);
    }
  }
}
```

```css
@import '@my/styles.css' layer(theme);
```

### What did you expect to happen?

Does not crash

### What actually happened?

TypeError: Cannot read properties of undefined (reading 'functionPartsStack')

Thrown from here: https://github.com/eslint/css/blob/268c7f03b981025e97c1489848212b5e9a27a9ab/src/rules/no-invalid-properties.js#L346

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/vitejs-vite-scw1hdur?file=src%2Fone.css

### Participation

- [ ] I am willing to submit a pull request for this issue.
